server/api/src/rs_get_resource_info_for_operation.cpp

```diff
@@ -13,31 +13,42 @@
 #include <cstring> // for strdup
 #include <string>
 
+namespace
+{
+    using log_api = irods::experimental::log::api;
+} // anonymous namespace
+
 int rs_get_resource_info_for_operation(rsComm_t* _rsComm, dataObjInp_t* _dataObjInp, char** _out_info)
 {
     using log_api = irods::experimental::log::api;
     if (_rsComm == nullptr || _dataObjInp == nullptr || _out_info == nullptr) {
-        return SYS_INVALID_INPUT_PARAM;
+        log_api::error("{}: Invalid input. Received one or more null pointers.", __func__);
+        return SYS_NULL_INPUT;
     }
     rodsServerHost_t* rodsServerHost = nullptr;
     const int remoteFlag = getAndConnRemoteZone(_rsComm, _dataObjInp, &rodsServerHost, REMOTE_OPEN);
     if (remoteFlag < 0) {
+        log_api::error("{}: getAndConnRemoteZone returned error: {}", __func__, remoteFlag);
         return remoteFlag;
     }
 
     if (REMOTE_HOST == remoteFlag) {
         return rc_get_resource_info_for_operation(rodsServerHost->conn, _dataObjInp, _out_info);
     }
 
+    const std::string message {"GET_RESOURCE_INFO_OP_TYPE_KW must be set to a valid operation."};
```

Comment:
I get the idea, but the places where this is used can be more explicit about what happened. You can append a statement to your existing one (via the logger) about receiving a null value or print the value that was passed by the client. That will help the admin a great deal.

As it is right now, this line can result in an unnecessary memory allocation for something that isn't used.

You can make it a compile-time string to avoid the allocation by using a C pointer or a `std::string_view`. For example:
```cpp
constexpr const char* message = "...";      // - or -
constexpr const char message[] = "...";     // - or -
constexpr std::string_view message = "..."; // Don't forget the header <string_view>
```

Reply:
Maybe we can talk about this one live.  I don't think I'm understanding the explanation ....

Reply:
Done too, I think. Not constexpr, but since the format is done  at runtime anyway, it'd probably be no savings....
LMK if this looks ok as is....

Reply:
Looks fine. Please mark as resolved.